v2/pkg/requests/bulk-http-request.go

```diff
@@ -121,39 +115,26 @@ func (r *BulkHTTPRequest) makeHTTPRequestFromModel(baseURL string, data string,
 	return &HttpRequest{Request: request}, nil
 }
 
-func (r *BulkHTTPRequest) StartGenerator() {
-	r.generatorChan = r.generator(r.basePayloads)
+func (r *BulkHTTPRequest) CreateGenerator(URL string) {
+	if r.gsfm == nil {
```

Comment:
Looks like this `r.gsfm` access should be protected as well?

Reply:
Good catch! I'm still moving things around, thanks for keeping an eye on this!